### ISSUE TITLE
gh-pages might be broken?

### ISSUE THREAD

**Issue by JacquesCarette** [JacquesCarette/Drasil#2365]
There was a `drasil-bot` that took care of updating the documentation pages and push them to the web site. This seems to have broken about 4 months ago (!).  It probably is related to Travis. Could you investigate?

**Comment by balacij** [comment]
It looks like`drasil-bot` is back up and running. I didn't touch it since this ticket creation (I don't think I have any access to it). Did you re-enable it?

We're just going to need to add a second workflow script that activates when something is pushed to `master` so as to trigger the gh-pages update. Related to: https://github.com/JacquesCarette/Drasil/issues/2287#issuecomment-829635814

**Comment by JacquesCarette** [comment]
I think `drasil-bot` has always been running, what is no longer running are the jobs to update the docs. These were hooked into Travis, *I think*. I'm not sure.  So yes, I think we need a new workflow for this.

**Comment by balacij** [comment]
Is it ok if I reset the [`.build-num`](https://github.com/JacquesCarette/Drasil/blob/gh-pages/.build-num) file on the `gh-pages` branch? It's currently aligned with Travis Builds, but we need it to align with GitHub Actions (unless we want to impose an offset to make up the difference).

For setting `drasil-bot` as the committer, do you have an personal access token secret setup on this repo for `drasil-bot`? If so, may I please have the name of the variable? I don't know if the `BOT_TOKEN` in the configuration files were only set on Travis or if they were also set on this repo's secrets.

**Comment by JacquesCarette** [comment]
Go ahead, please.  The personal access token is known to this repo as well.  I don't remember a 'variable' though.

I was looking at the settings today, and apparently the personal access token was done using an old protocol, and it would prefer if I were to regenerate it. So it seems like a good time to do so.  Please tell me (probably by linking to the right help pages) the process that I should follow.

**Comment by balacij** [comment]
Perfect, thank you!

We need to generate a [Personal Access Token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) with write permissions for `drasil-bot`, and then we would need to take the token and place it into a [repository secret](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository) (seems it could also go into an environment secret, but I don't think we would want a different bot per environment?), and then I would just need the name of secret so that our workflow script can access it.

**Comment by balacij** [comment]
I just got my testing bot account to [update](https://github.com/balacij/Drasil/commit/acbede65dc93f82f38b8e22760004e95296cf45b) my fork.

I gave the bot account collaborator access to my fork.

Also, I currently configured it for 3 secrets:
1. `BOT_TOKEN` containing the personal access token
2. `BOT_EMAIL` containing the email of the bot [email has to be on the list of emails registered on the bots account]
3. `BOT_NAME` containing the name (drasil-bot) of the bot

For the personal access token, I gave it the following permissions:
![image](https://user-images.githubusercontent.com/1627302/118285995-036a5780-b4a0-11eb-83c3-d0afeefe8f20.png)
